build-tools/packages/build-tools/src/type-test-generator/main.ts

```diff
@@ -62,23 +85,29 @@ const currentFile = new Project({
 
 const previousTsConfigPath = `${previousBasePath}/tsconfig.json`;
 let previousFile: SourceFile;
-if (existsSync(previousTsConfigPath)) {
-	const project = new Project({
-		skipFileDependencyResolution: true,
-		tsConfigFilePath: previousTsConfigPath,
-	});
-	previousFile = project.getSourceFileOrThrow("index.ts");
+const project = new Project({
+	skipFileDependencyResolution: true,
+	tsConfigFilePath: existsSync(previousTsConfigPath) ? previousTsConfigPath : undefined,
+});
+// Check for existence of alpha and add appropriate file
+if (existsSync(typeDefinitionFilePath)) {
+	project.addSourceFilesAtPaths(typeDefinitionFilePath);
+	previousFile = project.getSourceFileOrThrow("<package-name>-alpha.d.ts");
+	// Fall back to using .d.ts
 } else {
-	const project = new Project({
-		skipFileDependencyResolution: true,
-	});
 	project.addSourceFilesAtPaths(`${previousBasePath}/dist/**/*.d.ts`);
 	previousFile = project.getSourceFileOrThrow("index.d.ts");
 }
 
+/**
+ * Extracts type data from a TS source file and creates a map where each key is a type name and the value is its type data.
+ * @param file
+ * @returns Map<string, TypeData> mapping between item and its type
```

Comment:
The type info is already captured by the typescript signature, so we shouldn't need to denote it here.
I also always recommend complete sentence case wherever possible for accessibility.

```suggestion
 * @returns The mapping between item and its type
```